src/stridelayout.jl

```diff
@@ -573,54 +414,40 @@ end
     return Expr(:block, Expr(:meta, :inline), out)
 end
 
-"""
-    offsets(A) -> Tuple
-
-Returns offsets of indices with respect to 0. If values are known at compile time,
-it should return them as `Static` numbers.
-For example, if `A isa Base.Matrix`, `offsets(A) === (StaticInt(1), StaticInt(1))`.
-"""
-offsets(::Any) = (StaticInt{1}(),) # Assume arbitrary Julia data structures use 1-based indexing by default.
 @inline strides(A::Vector{<:Any}) = (StaticInt(1),)
 @inline strides(A::Array{<:Any,N}) where {N} = (StaticInt(1), Base.tail(Base.strides(A))...)
 @inline strides(A::AbstractArray) = _strides(A, Base.strides(A), contiguous_axis(A))
 
-@inline function strides(x::LinearAlgebra.Adjoint{T,V}) where {T,V<:AbstractVector{T}}
-    strd = stride(parent(x), One())
-    (strd, strd)
-end
-@inline function strides(x::LinearAlgebra.Transpose{T,V}) where {T,V<:AbstractVector{T}}
-    strd = stride(parent(x), One())
-    (strd, strd)
+function strides(x::VecAdjTrans)
+    p = parent(x)
+    st = first(strides(p))
+    return (static_length(p) * st, st)
```

Comment:
Is this change alright? I came across it while trying to improve coverage. The previous method just returned `(st, st)` but this one has the same results as Base.

Reply:
I had it that way to simplify indexing code.
Unlike linearly indexing a transposed matrix `A'[i]` where the index increments the pointer by a multiple of `stride(A's, 1)`, if it's a vector we increment by `stride(a', 2)`.
So we'd have to special case the indexing code, or just return that index here.
Unless that vector is a view of a matrix and you're deliberately accessing it out of bounds to access a neighboring vector, I don't know why you'd want the behavior of the base stride function.

Reply:
That makes sense. I'll revert that change and fix the tests I've added for it.